hs_composite_resource/models.py

```diff
@@ -333,6 +345,38 @@ def get_aggregation_by_name(self, name, aggregations=None):
             raise ObjectDoesNotExist(
                 "No matching aggregation was found for name:{}".format(name))
 
+    def get_aggregation_by_meta_file(self, meta_file_path: str) -> AggregationType:
+        """Get an aggregation that matches the specified meta xml/json file path
+        :param  meta_file_path: directory path of the meta xml/json file
+        :return an aggregation object if found
+        :raises ObjectDoesNotExist if no matching aggregation is found
+        """
+        if __debug__:
+            assert any(meta_file_path.endswith(ext) for ext in AggregationMetaFilePath)
+
+        meta_file_path = self.get_relative_path(meta_file_path)
+        folder, base = os.path.split(meta_file_path)
+        if base.endswith(AggregationMetaFilePath.METADATA_FILE_ENDSWITH):
+            base_file_name_to_match = base[:-len(AggregationMetaFilePath.METADATA_FILE_ENDSWITH)]
+        elif base.endswith(AggregationMetaFilePath.RESMAP_FILE_ENDSWITH):
+            base_file_name_to_match = base[:-len(AggregationMetaFilePath.RESMAP_FILE_ENDSWITH)]
+        else:
+            base_file_name_to_match = base[:-len(AggregationMetaFilePath.SCHEMA_JSON_FILE_ENDSWITH)]
+
+        for res_file in ResourceFile.list_folder(self, folder=folder, sub_folders=False):
+            if res_file.has_logical_file:
+                file_name_to_match = f"{base_file_name_to_match}{res_file.extension}"
+                if res_file.file_name == file_name_to_match:
+                    return res_file.logical_file
+        # check for folder aggregation
+        if folder and folder == base_file_name_to_match:
+            folder_aggregation = self.get_folder_aggregation_object(folder)
+            if folder_aggregation is not None:
+                return folder_aggregation
```

Comment:
in one case we return a logical file, in the other case we return a folder_aggregation. Don't we expect to always return an aggregation?

Reply:
logical_file is same as aggregation. In the database it is logical_file.

Reply:
I see it's either a folder aggregation or a logicalFile aggregation. Thanks!